enhancements/declarative-index-config.md

```diff
@@ -0,0 +1,483 @@
+        "bundles": [
+            {
+                "path": "quay.io/random/etcdv0.0.1",
+                "version": "v0.0.1",
+                "channel": "alpha",
+                "upgradeMode": "semver",
+                "ownedAPIs": [
+                    {
+                        "group": "etcd.database.coreos.com",
+                        "version": "v1alpha1",
+                        "kind": "etcd"
+                    }
+                ],
+                "requiredAPIs": []
+            },
+            {
+                "path": "quay.io/somethingelse/etcdv0.0.2",
+                "version": "v0.0.2",
+                "channel": "alpha",
+                "upgradeMode": "semver",
+                "ownedAPIs": [
+                    {
+                        "group": "etcd.database.coreos.com",
+                        "version": "v1alpha1",
+                        "kind": "etcd"
+                    }
+                ],
+                "requiredAPIs": []
+            }
+        ]
+    }
+```
+
+#### Story 3
+
+As an index author(with push permission to the namespace my index is hosted in), I can edit the structure of my index (remove bundles from index, change channel bundle is in), using only the json representation of the index/packages inside the index. 
+
+```bash=
+$ cd community-operators 
+$ //TODO come up with sed command to edit "channel":"alpha" to "channel":"beta" for one of the bundles in etcd.json 
+$ cd ..
+$ opm index update community-operators --tag=docker.io/my-namespace/community-operators:latest
+$ rm -rf community-operators
+$ opm index represent packages --index=docker.io/my-namespace/community-operators --output=json
+$ cat community-operators/etcd.json
+"etcd": {
+        "name": "etcd",
+        "defaultChannel": "alpha",
+        "icon": "asdfasdfoipuasdfvasdfqer",
+        "bundles": [
+            {
+                "path": "quay.io/random/etcdv0.0.1",
+                "version": "v0.0.1",
+                "channel": "alpha",
+                "upgradeMode": "semver",
+                "ownedAPIs": [
+                    {
+                        "group": "etcd.database.coreos.com",
+                        "version": "v1alpha1",
+                        "kind": "etcd"
+                    }
+                ],
+                "requiredAPIs": []
+            },
+            {
+                "path": "quay.io/somethingelse/etcdv0.0.2",
+                "version": "v0.0.2",
+                "channel": "beta",
+                "upgradeMode": "semver",
+                "ownedAPIs": [
+                    {
+                        "group": "etcd.database.coreos.com",
+                        "version": "v1alpha1",
+                        "kind": "etcd"
+                    }
+                ],
+                "requiredAPIs": []
+            }
+        ]
+    }
+
+```
+
+Similarly, removing a `.json` from `community-operators` and pushing that update upto the container image removes the bundle etc. 
```

Comment:
is the assumption here that whatever i specify in the json overrides whatever exists in the index upgrade db?

So in order to do this i need to ensure that my json is based on the latest version of the index db?  (otherwise i risk stomping on changes)

Reply:
@bparees that's a good point, updated with this safety check[1,2] 

[1] https://github.com/operator-framework/enhancements/pull/37/files#diff-1edbfba9fdddd2b74b5cd32c6de146a4f62e1400a93bee78aed2cbf6ebaaa1d7R297
[2] https://github.com/operator-framework/enhancements/pull/37/files#diff-1edbfba9fdddd2b74b5cd32c6de146a4f62e1400a93bee78aed2cbf6ebaaa1d7R315